### PR TITLE
Windows: clean up building and installer

### PR DESCRIPTION
**Pull request by Anakonda**
Mainly moved all the stdlib .py-files into one zip. (#983)
.dll-files can not be in the zip, moved them in ´bin´. Exe and all necessary dlls in main folder.

### CODE REVIEW

File: deployment/windows/buildwin32.py [hunk 1]
```diff
@@ -98,23 +109,30 @@ def finddlls(exe):
...
+        shutil.copyfile(file, os.path.join(builddir, 'bin', relpath))
+    else:
+        relativepath = file[file.find('lib')+14:]
+        if 'encodings' in file:
+            continue
+        if relativepath not in library.namelist():
+            if relativepath.startswith('site-packages'):
+                relativepath = relativepath[14:]
+            library.write(file, relativepath)
     if file[-4:] == '.dll' or file[-4:] == '.DLL':
```
Comment (ThomasWaldmann) [inline deployment/windows/buildwin32.py]: file.endswith(('.dll', '.DLL'))

File: deployment/windows/buildwin32.py [hunk 2]
```diff
@@ -98,23 +109,30 @@ def finddlls(exe):
...
-        os.makedirs(os.path.join(builddir, 'lib', 'python3.5', os.path.split(relpath)[0]), exist_ok=True)
-        shutil.copyfile(file, os.path.join(builddir, 'lib', 'python3.5', relpath))
-        continue
-    relativepath = file[file.find('lib')+4:]
-    os.makedirs(os.path.join(builddir, 'lib', os.path.split(relativepath)[0]), exist_ok=True)
-    shutil.copyfile(file, os.path.join(builddir, 'lib', relativepath))
+        os.makedirs(os.path.join(builddir, 'bin', os.path.split(relpath)[0]), exist_ok=True)
+        shutil.copyfile(file, os.path.join(builddir, 'bin', relpath))
+    else:
+        relativepath = file[file.find('lib')+14:]
```
Comment (ThomasWaldmann) [inline deployment/windows/buildwin32.py]: +14 ?

File: deployment/windows/buildwin32.py [hunk 3]
```diff
@@ -98,23 +109,30 @@ def finddlls(exe):
...
+        if 'encodings' in file:
+            continue
+        if relativepath not in library.namelist():
+            if relativepath.startswith('site-packages'):
+                relativepath = relativepath[14:]
```
Comment (ThomasWaldmann) [inline deployment/windows/buildwin32.py]: another 14.

File: deployment/windows/buildwin32.py [hunk 4]
```diff
@@ -98,23 +109,30 @@ def finddlls(exe):
...
-                shutil.copyfile(dll, os.path.join(builddir, 'bin', os.path.split(dll)[1]))
-for dll in finddlls(os.path.join(builddir, "bin/borg.exe")):
+                shutil.copyfile(dll, os.path.join(builddir, os.path.split(dll)[1]))
+for dll in finddlls(os.path.join(builddir, "borg.exe")):
     if builddir not in dll:
-        shutil.copyfile(dll, os.path.join(builddir, 'bin', os.path.split(dll)[1]))
+        shutil.copyfile(dll, os.path.join(builddir, os.path.split(dll)[1]))
 
-shutil.copyfile('src/borg/__main__.py', os.path.join(builddir, 'lib/python3.5/borg/__main__.py'))
+shutil.copyfile('src/borg/__main__.py', os.path.join(builddir, 'bin', 'borg/__main__.py'))
```
Comment (ThomasWaldmann) [inline deployment/windows/buildwin32.py]:
```
..., 'borg', '__main__.py')
```
And strictly taken, the 1st param also should use o.p.j.

**Comment by enkore** [comment]
Nice, thanks. It not only installs much faster, but also starts faster. I added a tiny fixup in #1176

**Pull request merged**